internal/flags/apiflags/flags.go

```diff
@@ -0,0 +1,152 @@
+	GitInit          bool
+	MakeVendor       bool
+	SkipValidation   bool
+	CrdVersion       string
+
+	HelmChartRef     string
+	HelmChartVersion string
+	HelmChartRepo    string
+}
+
+// AddTo - Add the reconcile period and watches file flags to the the flagset
+// helpTextPrefix will allow you add a prefix to default help text. Joined by a space.
+func (f *AddAPIFlags) AddTo(flagSet *pflag.FlagSet) {
+	flagSet.StringVar(&f.APIVersion,
+		"api-version",
+		"",
+		"Kubernetes apiVersion and has"+
+			" a format of $GROUP_NAME/$VERSION (e.g app.example.com/v1alpha1) - used with \"ansible\" or \"helm\" types")
+	flagSet.StringVar(&f.Kind,
+		"kind",
+		"",
+		"Kubernetes resource Kind name. (e.g AppService)")
+	flagSet.StringVar(&f.OperatorType,
+		"type",
+		"go",
+		"Type of operator to initialize (choices: \"go\", \"ansible\" or \"helm\")")
+	flagSet.StringVar(&f.Repo,
+		"repo",
+		"",
+		"Project repository path for Go operators. Used as the project's Go import path. This must be set if "+
+			"outside of $GOPATH/src (e.g. github.com/example-inc/my-operator)")
+	flagSet.BoolVar(&f.GitInit,
+		"git-init",
+		false,
+		"Initialize the project directory as a git repository (default false)")
+	flagSet.StringVar(&f.HeaderFile,
+		"header-file",
+		"",
+		"Path to file containing headers for generated Go files. Copied to hack/boilerplate.go.txt")
+	flagSet.BoolVar(&f.MakeVendor,
+		"vendor",
+		false,
+		"Use a vendor directory for dependencies")
+	flagSet.BoolVar(&f.SkipValidation,
+		"skip-validation",
+		false,
+		"Do not validate the resulting project's structure and dependencies. (Only used for --type go)")
+	flagSet.BoolVar(&f.GeneratePlaybook, "generate-playbook", false,
+		"Generate a playbook skeleton. (Only used for --type ansible)")
+
+	flagSet.BoolVar(&f.SkipGeneration,
+		"skip-generation",
+		false,
+		"Skip generation of deepcopy and OpenAPI code and OpenAPI CRD specs")
+	flagSet.StringVar(&f.CrdVersion,
+		"crd-version",
+		gencrd.DefaultCRDVersion,
+		"CRD version to generate")
+	flagSet.StringVar(&f.HelmChartRef,
+		"helm-chart",
+		"",
+		"Initialize helm operator with existing helm chart (<URL>, <repo>/<name>, or local path)")
+	flagSet.StringVar(&f.HelmChartVersion,
+		"helm-chart-version",
+		"",
+		"Specific version of the helm chart (default is latest version)")
+	flagSet.StringVar(&f.HelmChartRepo,
+		"helm-chart-repo",
+		"",
+		"Chart repository URL for the requested helm chart")
+
+}
+
+// VerifyFlags ...
+func (f *AddAPIFlags) VerifyFlags() error {
+	if f.OperatorType != projutil.OperatorTypeGo && f.OperatorType != projutil.OperatorTypeAnsible && f.OperatorType !=
+		projutil.OperatorTypeHelm {
+		return fmt.Errorf("value of --type can only be `go`, `ansible`, or `helm`: %v",
+			projutil.ErrUnknownOperatorType{Type: f.OperatorType})
+	}
+	if f.OperatorType != projutil.OperatorTypeAnsible && f.GeneratePlaybook {
+		return fmt.Errorf("value of --generate-playbook can only be used with --type `ansible`")
+	}
+
```

Comment:
IHMO we should encapsulated the check by types. So it would be:
- Check all that is common
- then validateHelmFlags
- then validateAnsible ones.

Reply:
Improvement: It seems to me, this is an improvement being asked for existing VerifyFlags() 
@joelanford WDYT?

Reply:
@camilamacedo86 @jmrodri  Will add above refactoring to the Github Issue as well, as stated here. https://github.com/operator-framework/operator-sdk/pull/2703#discussion_r400488640

Reply:
This is basically the logic from the existing function in `new`, right @bharathi-tenneti? If so, it isn't quite in-scope to do @camilamacedo86's suggested improvement in this PR. Let's do it in a follow-up.